pages/api/tokenData/static.ts

```diff
@@ -0,0 +1,109 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { ethers } from 'ethers';
+import { HASH_ATTRIBUTES } from '../../../util/hash_attributes';
+import {
+  hex2bin,
+  getHashesContract
+} from '../../../util';
+
+type BinaryAttribute = { trait_type: string, value: number };
+
+type ResponseData = {
+  hash: string
+  binary_value: string
+  binary_attributes: BinaryAttribute[]
+  type: string
+  phrase_value: string
+  phrase_attributes: BinaryAttribute[]
+};
+
+function getHashBinaryAttributes(hash: string): BinaryAttribute[] {
+  function getAttrValue(attr: Attribute, hash: string) {
+    if (attr.regex) {
+      const hex2binVal = hex2bin(hash).match(attr.regex);
+      const binaryAttrLength = hex2binVal !== null ? hex2binVal[0].length : 0;
+      return binaryAttrLength;
+    }
+
+    if (attr.calculationFunction) return attr.calculationFunction(hash);
+
+    return null;
+  }
+
+  return HASH_ATTRIBUTES.map((attribute) => ({
+    trait_type: attribute.descriptionShort,
+    value: getAttrValue(attribute, hash)
+  }));
+}
+
+function getPhraseRarity(str: string, arr: ethers.Event[]): number {
+  const allPhrases = arr.map(event => event?.args?.phrase);
+  const phraseAmount = allPhrases.filter((phrase: string) => phrase === str).length;
+  return phraseAmount / allPhrases.length * 100;
+}
+
+function getPhraseAttributes(
+  phrase: string,
+  events: ethers.Event[]
+): BinaryAttribute[] {
+  const phraseWords: string[] = phrase.split(' ');
+  const phraseCharCount: number = phraseWords.reduce((prev: number, curr: string) => prev + curr.length, 0);
+  return [
+    {
+      trait_type: 'rarity',
+      value: getPhraseRarity(phrase, events)
+    },
+    {
+      trait_type: 'word amount',
+      value: phraseWords.length
+    },
+    {
+      trait_type: 'character amount',
+      value: phraseCharCount
+    }
+  ];
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ResponseData | string>
+) {
+  const { tokenId } = req.query;
+
+  if (isNaN(Number(tokenId))) {
+    res.status(400).send('tokenId must be a number');
+    return;
+  }
+
+  const hashesContract = getHashesContract(1);
```

Comment:
you may want to get the 'nonce' from the hashes contract and check if the tokenId is greater than that. If it is, return early with a 404 or 400